### PR TITLE
11 UI first script

### LINKED ISSUE #11: UI First Screen
After #12 

image

Ne intereseaza sa cream primul screen de UI, cel de `Boards`.
Cea mai usoara metoda (din cate am lucrat si eu cu Kivy) este:
1. Realizeaza un prototip static in fiserul .kv (cu unul, doua, trei board-uri fictive)
2. Aranjeaza pana iti place cum arata
3. Sterge toate board-urile fictive, dar pastreaza layout-ul parinte
4. Defineste o clasa pentru widget-ul de Board
5. Dupa ce porneste aplicatia, extrage toate `Board` din `App` si creaza o instanta (din clasa definita mai sus) pentru fiecare board si adauga-le in layout

Directii generale:
- Foloseste un [BoxLayout](https://kivy.org/doc/stable/api-kivy.uix.boxlayout.html) ca si layout pentru Board-uri
- Nu uita ca imaginea trebuie sa fie clickable
- Nu uita de butonul de la final pentru a crea un alt board
- Cauta pe net tot ce ai nevoie, sunt destule materiale

### PR DESCRIPTION
**Pull request by mariaaliu**

### CODE REVIEW

**Comment by mariaaliu** [comment]
Am incercat zilele astea sa fac scrollable lista de records, dar nu reusesc sa imi dau seama ce sa fac...

**Comment by mariaaliu** [comment]
Nu ma prind cum sa "salvez" boardul in care sunt ca sa pot sa iau records si tasks

**Comment by DanBrezeanu** [comment]
> Nu ma prind cum sa "salvez" boardul in care sunt ca sa pot sa iau records si tasks

Cand [creezi](https://github.com/mariaaliu/HiPlan/blob/8955c27f0428b296b612dc4feb1435fe0b5d3be4/hiplan/ui/app.py#L31) un `BoardIcon`, ar trebui sa-i pasezi (prin `kwargs` evident) si `Board`-ul (din backend) asociat. 

```
self.ids.boards_layout.add_widget(
      BoardIcon(
          text=board.name,
          background=board.background,
          board=board,
)
```

Evident, `BoardIcon` ar trebui sa-l pastreze ca un atribut (`self.board = kwargs.pop['board']`)

Acum, cand dai click pe un `BoardIcon` ai vrea sa se deschida un `BoardScreen` cu `Record`-urile asociate.

Cum am vorbit si data trecuta `BoardScreen` trebuie sa aiba o referinta la `Board`-ul asociat. De unde il luam?
Evident, din `BoardIcon`. 
Da-i lui `BoardScreen` sa ia un `Board` ca si parametri (identic cu ce am facut la `BoardIcon`).

Cand vrem sa se deschida un `BoardScreen`? Cand se apasa pe un `BoardIcon`. Deci de fiecare data cand creezi un `BoardIcon`, fa-i si `bind`. Ce ai tu de facut acum este sa-ti dai seama care ar trebui sa fie actiunea care se executa cand se apasa pe buton.

**Comment by mariaaliu** [comment]
Am eroarea asta:
Traceback (most recent call last):
   File "C:\Users\marin\HiPlan\hiplan\ui\app.py", line 141, in <module>
     HiPlanApp().run()
   File "C:\Users\marin\HiPlan\env\lib\site-packages\kivy\app.py", line 949, in run
     self._run_prepare()
   File "C:\Users\marin\HiPlan\env\lib\site-packages\kivy\app.py", line 919, in _run_prepare
     root = self.build()
   File "C:\Users\marin\HiPlan\hiplan\ui\app.py", line 134, in build
     self.screen_manager.add_widget(screen=BoardScreen(name='board-screen'))
   File "C:\Users\marin\HiPlan\hiplan\ui\app.py", line 68, in _init_
     super()._init_(**kwargs)
   File "C:\Users\marin\HiPlan\env\lib\site-packages\kivy\uix\relativelayout.py", line 274, in _init_
     super(RelativeLayout, self)._init_(**kw)
   File "C:\Users\marin\HiPlan\env\lib\site-packages\kivy\uix\floatlayout.py", line 65, in _init_
     super(FloatLayout, self)._init_(**kwargs)
   File "C:\Users\marin\HiPlan\env\lib\site-packages\kivy\uix\layout.py", line 76, in _init_
     super(Layout, self)._init_(**kwargs)
   File "C:\Users\marin\HiPlan\env\lib\site-packages\kivy\uix\widget.py", line 359, in _init_
     self.apply_class_lang_rules(
   File "C:\Users\marin\HiPlan\env\lib\site-packages\kivy\uix\widget.py", line 463, in apply_class_lang_rules
     Builder.apply(
   File "C:\Users\marin\HiPlan\env\lib\site-packages\kivy\lang\builder.py", line 541, in apply
     self._apply_rule(
   File "C:\Users\marin\HiPlan\env\lib\site-packages\kivy\lang\builder.py", line 663, in _apply_rule
     self._apply_rule(
   File "C:\Users\marin\HiPlan\env\lib\site-packages\kivy\lang\builder.py", line 663, in _apply_rule
     self._apply_rule(
   File "C:\Users\marin\HiPlan\env\lib\site-packages\kivy\lang\builder.py", line 663, in _apply_rule
     self._apply_rule(
   [Previous line repeated 1 more time]
   File "C:\Users\marin\HiPlan\env\lib\site-packages\kivy\lang\builder.py", line 659, in _apply_rule
     child = cls(__no_builder=True)
 TypeError: _init_() missing 1 required positional argument: 'title'

Si nu prea imi dau seama ce spune...

File: hiplan/ui/app.py [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+        super().__init__(**kwargs)
+        boards = App.get_running_app().app.boards
+        for board in boards:
+            button = BoardIcon(
+                    text=board.name,
+                    background=board.background,
+                    board=board
+                )
+            self.ids.boards_layout.add_widget(button)
+            button.bind(on_press = self.callback)
```
Comment (DanBrezeanu) [inline hiplan/ui/app.py:49]: nu exista `self.callback`

File: hiplan/ui/app.py [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+                    record = record
+                )
+            )
+            tasks = record.tasks
+            for task in tasks:
+                self.ids.task_layout.add_widget(
+                    TwoLineListItem(
+                        text = task.title,
+                        secondary_text = task.description,
+                        task = task
```
Comment (DanBrezeanu) [inline hiplan/ui/app.py]: `TwoLineListItem` nu ia un `task` ca si `kwargs`

**Comment by mariaaliu** [comment]
Nu imi dau seama cum trebuie sa deschid un BoardScreen dupa ce apas pe un BoardIcon

**Comment by DanBrezeanu** [comment]
> Nu imi dau seama cum trebuie sa deschid un BoardScreen dupa ce apas pe un BoardIcon

Ii dai la `on_press`-ul de la `bind` o functie care creeaza un `BoardScreen` si il si seteaza ca si `.current` pentru `ScreenManager` bazat pe datele gasite in `BoardIcon` si backend.

**Comment by mariaaliu** [comment]
Am o problema cu MDTextFieldRound(), pentru ca nu ia kwargs si nici clasele care il mostenesc nu pot lua:
(**kwargs) -> None
The base class of the class hierarchy.

When called, it accepts no arguments and returns a new featureless instance that has no instance attributes and cannot be given any.

**Comment by DanBrezeanu** [comment]
> Am o problema cu MDTextFieldRound(), pentru ca nu ia kwargs si nici clasele care il mostenesc nu pot lua:
> (**kwargs) -> None
> The base class of the class hierarchy.
> 
> When called, it accepts no arguments and returns a new featureless instance that has no instance attributes and cannot be given any.

In [documentatie](https://kivymd.readthedocs.io/en/latest/components/text-field/index.html#kivymd.uix.textfield.MDTextFieldRound) pare ca ia `kwargs`. Care e problema mai exact?

**Comment by mariaaliu** [comment]
Daca incerc sa scriu (in python):

  task_deadline = TaskDeadline(multiline=False, callback=self.update_deadline)
  self.ids.task_screen.add_widget(task_deadline)
  
  task_progress_status = MDTextFieldRound(multiline= True, hint_text= 'Progress: SOLVED / UNSOLVED ?', size_hint_y = None)
  self.ids.task_screen.add_widget(task_progress_status)

In loc de (in kivy):

                TaskDeadline:
                    id: task_deadline
                    multiline: False
                    hint_text: 'Deadline'
                    size_hint_y: None
                
                MDTextFieldRound:
                    id: task_progress_status
                    multiline: True
                    hint_text: 'Progress: SOLVED / UNSOLVED ?'
                    size_hint_y: None
                    pos_hint_y: {'top': 1} 

Am eroarea asta:
.
.
.
Screenshot 2021-05-14 181302

File "C:\Users\marin\HiPlan\hiplan\ui\app.py", line 139, in callback
     self.screen_manager.add_widget(screen=TaskScreen(task=instance.task, name=instance.task.title))
   File "C:\Users\marin\HiPlan\hiplan\ui\app.py", line 170, in __init__
     task_deadline = TaskDeadline(multiline=False, callback=self.update_deadline)
   File "C:\Users\marin\HiPlan\hiplan\ui\app.py", line 150, in __init__
     super().__init__(**kwargs)
   File "C:\Users\marin\HiPlan\env\lib\site-packages\kivymd\uix\textfield.py", line 1193, in __init__
     super().__init__(**kwargs)
   File "C:\Users\marin\HiPlan\env\lib\site-packages\kivymd\theming.py", line 907, in __init__
     super().__init__(**kwargs)
   File "C:\Users\marin\HiPlan\env\lib\site-packages\kivy\uix\textinput.py", line 528, in __init__
     super(TextInput, self).__init__(**kwargs)
   File "C:\Users\marin\HiPlan\env\lib\site-packages\kivy\uix\behaviors\focus.py", line 367, in __init__
     super(FocusBehavior, self).__init__(**kwargs)
   File "C:\Users\marin\HiPlan\env\lib\site-packages\kivy\uix\widget.py", line 350, in __init__
     super(Widget, self).__init__(**kwargs)
   File "kivy\_event.pyx", line 245, in kivy._event.EventDispatcher.__init__
 TypeError: object.__init__() takes exactly one argument (the instance to initialize)

**Comment by DanBrezeanu** [comment]
Eroarea ta pare ca vine de la `TaskDeadline`, nu de la `TextField`. Banuiala mea este ca ii dai acel parametru `callback` pe care uiti sa-l scoti din `kwargs` cand apelezi `super.__init__`

**Comment by mariaaliu** [comment]
Cred ca am rezolvat. Am facut si deadline sa completeze automat campul cu data aleasa. Acum ma ocup de butoane

**Comment by mariaaliu** [comment]
Am cam terminat cu screen-urile. Nu toate butoanele fac ceva, dat cele care duc de la un screen la altul si cele care te duc inapoi sunt setate.
Nu stiu exact de ce Some checks were not successful:
ERROR: Could not find a version that satisfies the requirement kivy-deps.angle==0.3.0 (from versions: none)
ERROR: No matching distribution found for kivy-deps.angle==0.3.0

**Pull request merged**